SecurityException not caught from OmniataEventWorker#sendEvent (HttpURLConnection#getInputStream)

We can see crash reports where a `SecurityException` is thrown from the `OmniataEventWorker #sendEvent` method. Since we cannot handle this exception in our client code it should be handled by the SDK instead. Here is the stack traces from the exception:

```
java.lang.SecurityException: Permission denied (missing INTERNET permission?)
       at java.net.InetAddress.lookupHostByName(InetAddress.java:429)
       at java.net.InetAddress.getAllByNameImpl(InetAddress.java:247)
       at java.net.InetAddress.getAllByName(InetAddress.java:214)
       at libcore.net.http.HttpConnection.(HttpConnection.java:70)
       at libcore.net.http.HttpConnection.(HttpConnection.java:50)
       at libcore.net.http.HttpConnection$Address.connect(HttpConnection.java:341)
       at libcore.net.http.HttpConnectionPool.get(HttpConnectionPool.java:87)
       at libcore.net.http.HttpConnection.connect(HttpConnection.java:128)
       at libcore.net.http.HttpEngine.openSocketConnection(HttpEngine.java:315)
       at libcore.net.http.HttpsURLConnectionImpl$HttpsEngine.makeSslConnection(HttpsURLConnectionImpl.java:461)
       at libcore.net.http.HttpsURLConnectionImpl$HttpsEngine.connect(HttpsURLConnectionImpl.java:433)
       at libcore.net.http.HttpEngine.sendSocketRequest(HttpEngine.java:289)
       at libcore.net.http.HttpEngine.sendRequest(HttpEngine.java:239)
       at libcore.net.http.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:273)
       at libcore.net.http.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:168)
       at libcore.net.http.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:271)
       at com.omniata.android.sdk.OmniataEventWorker.sendEvent(OmniataEventWorker.java:165)
       at com.omniata.android.sdk.OmniataEventWorker.processEvents(OmniataEventWorker.java:116)
       at com.omniata.android.sdk.OmniataEventWorker.run(OmniataEventWorker.java:91)
       at java.lang.Thread.run(Thread.java:856)
Caused by: libcore.io.GaiException: getaddrinfo failed: EAI_NODATA (No address associated with hostname)
       at libcore.io.Posix.getaddrinfo(Posix.java)
       at libcore.io.ForwardingOs.getaddrinfo(ForwardingOs.java:55)
       at java.net.InetAddress.lookupHostByName(InetAddress.java:416)
       at java.net.InetAddress.getAllByNameImpl(InetAddress.java:247)
       at java.net.InetAddress.getAllByName(InetAddress.java:214)
       at libcore.net.http.HttpConnection.(HttpConnection.java:70)
       at libcore.net.http.HttpConnection.(HttpConnection.java:50)
       at libcore.net.http.HttpConnection$Address.connect(HttpConnection.java:341)
       at libcore.net.http.HttpConnectionPool.get(HttpConnectionPool.java:87)
       at libcore.net.http.HttpConnection.connect(HttpConnection.java:128)
       at libcore.net.http.HttpEngine.openSocketConnection(HttpEngine.java:315)
       at libcore.net.http.HttpsURLConnectionImpl$HttpsEngine.makeSslConnection(HttpsURLConnectionImpl.java:461)
       at libcore.net.http.HttpsURLConnectionImpl$HttpsEngine.connect(HttpsURLConnectionImpl.java:433)
       at libcore.net.http.HttpEngine.sendSocketRequest(HttpEngine.java:289)
       at libcore.net.http.HttpEngine.sendRequest(HttpEngine.java:239)
       at libcore.net.http.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:273)
       at libcore.net.http.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:168)
       at libcore.net.http.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:271)
       at com.omniata.android.sdk.OmniataEventWorker.sendEvent(OmniataEventWorker.java:165)
       at com.omniata.android.sdk.OmniataEventWorker.processEvents(OmniataEventWorker.java:116)
       at com.omniata.android.sdk.OmniataEventWorker.run(OmniataEventWorker.java:91)
       at java.lang.Thread.run(Thread.java:856)
Caused by: libcore.io.ErrnoException: getaddrinfo failed: EACCES (Permission denied)
       at libcore.io.Posix.getaddrinfo(Posix.java)
       at libcore.io.ForwardingOs.getaddrinfo(ForwardingOs.java:55)
       at java.net.InetAddress.lookupHostByName(InetAddress.java:416)
       at java.net.InetAddress.getAllByNameImpl(InetAddress.java:247)
       at java.net.InetAddress.getAllByName(InetAddress.java:214)
       at libcore.net.http.HttpConnection.(HttpConnection.java:70)
       at libcore.net.http.HttpConnection.(HttpConnection.java:50)
       at libcore.net.http.HttpConnection$Address.connect(HttpConnection.java:341)
       at libcore.net.http.HttpConnectionPool.get(HttpConnectionPool.java:87)
       at libcore.net.http.HttpConnection.connect(HttpConnection.java:128)
       at libcore.net.http.HttpEngine.openSocketConnection(HttpEngine.java:315)
       at libcore.net.http.HttpsURLConnectionImpl$HttpsEngine.makeSslConnection(HttpsURLConnectionImpl.java:461)
       at libcore.net.http.HttpsURLConnectionImpl$HttpsEngine.connect(HttpsURLConnectionImpl.java:433)
       at libcore.net.http.HttpEngine.sendSocketRequest(HttpEngine.java:289)
       at libcore.net.http.HttpEngine.sendRequest(HttpEngine.java:239)
       at libcore.net.http.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:273)
       at libcore.net.http.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:168)
       at libcore.net.http.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:271)
       at com.omniata.android.sdk.OmniataEventWorker.sendEvent(OmniataEventWorker.java:165)
       at com.omniata.android.sdk.OmniataEventWorker.processEvents(OmniataEventWorker.java:116)
       at com.omniata.android.sdk.OmniataEventWorker.run(OmniataEventWorker.java:91)
       at java.lang.Thread.run(Thread.java:856)
```

--- a/src/com/omniata/android/sdk/OmniataEventWorker.java
+++ b/src/com/omniata/android/sdk/OmniataEventWorker.java
@@ -207,6 +207,9 @@ class OmniataEventWorker implements Runnable {
 		} catch (IOException e) {
 			OmniataLog.e(TAG, e.toString());
 			return EventStatus.RETRY;
+		} catch (SecurityException e) {
+			OmniataLog.e(TAG, e.toString());
+			return EventStatus.RETRY;
 		} finally {
 			if (connection != null) {
 				connection.disconnect();

